extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractor.java

```diff
@@ -0,0 +1,175 @@
+    }
+
+    @Override
+    public String getUploaderUrl() {
+        //Youtube mix are auto-generated
+        return "";
+    }
+
+    @Override
+    public String getUploaderName() {
+        //Youtube mix are auto-generated by YouTube
+        return "YouTube";
+    }
+
+    @Override
+    public String getUploaderAvatarUrl() {
+        //Youtube mix are auto-generated by YouTube
+        return "https://upload.wikimedia.org/wikipedia/commons/thumb/0/09/YouTube_full-color_icon_(2017).svg/320px-YouTube_full-color_icon_(2017).svg.png";
+    }
+
+    @Override
+    public long getStreamCount() {
+        // Auto-generated playlist always start with 25 videos and are endless
+        return ListExtractor.ITEM_COUNT_INFINITE;
+    }
+
+    @Nonnull
+    @Override
+    public InfoItemsPage<StreamInfoItem> getInitialPage() throws ExtractionException {
+        StreamInfoItemsCollector collector = new StreamInfoItemsCollector(getServiceId());
+        collectStreamsFrom(collector, playlistData.getArray(CONTENTS));
+        return new InfoItemsPage<>(collector, getNextPageUrl());
+    }
+
+    @Override
+    public String getNextPageUrl() throws ExtractionException {
+        final JsonObject lastStream = ((JsonObject) playlistData.getArray(CONTENTS)
+            .get(playlistData.getArray(CONTENTS).size() - 1));
+        if (lastStream == null || lastStream.getObject(PLAYLIST_PANEL_VIDEO_RENDERER) == null) {
+            throw new ExtractionException("Could not extract next page url");
+        }
+        return "https://youtube.com" + lastStream.getObject(PLAYLIST_PANEL_VIDEO_RENDERER)
+            .getObject("navigationEndpoint").getObject("commandMetadata")
+            .getObject("webCommandMetadata").getString("url") + "&pbj=1";
+    }
+
+    @Override
+    public InfoItemsPage<StreamInfoItem> getPage(final String pageUrl)
+            throws ExtractionException, IOException {
+        if (pageUrl == null || pageUrl.isEmpty()) {
+            throw new ExtractionException(
+                new IllegalArgumentException("Page url is empty or null"));
+        }
+
+        StreamInfoItemsCollector collector = new StreamInfoItemsCollector(getServiceId());
+        final JsonArray ajaxJson = getJsonResponse(pageUrl, getExtractorLocalization());
+        playlistData =
+            ajaxJson.getObject(3).getObject(RESPONSE).getObject(CONTENTS)
+                .getObject(TWO_COLUMN_WATCH_NEXT_RESULTS).getObject(PLAYLIST)
+                .getObject(PLAYLIST);
+        final JsonArray streams = playlistData.getArray(CONTENTS);
+        //Because continuation requests are created with the last video of previous request as start
+        streams.remove(0);
+        collectStreamsFrom(collector, streams);
+        return new InfoItemsPage<>(collector, getNextPageUrl());
+    }
+
+    private void collectStreamsFrom(
+            @Nonnull StreamInfoItemsCollector collector,
+            @Nullable JsonArray streams) {
+        collector.reset();
```

Comment:
You don't need to call `collector.reset()` here as every time you call it, you already have a new empty collector.